server.js

```diff
@@ -39,23 +33,21 @@ const {subscribeToInvoices} = require('./push');
 const {subscribeToTransactions} = require('./push');
 const {transactionsRouter} = require('./routers');
 const {unlockWallet} = require('./lightning');
-const {verifyClient} = require('./push');
 const {walletInfoRouter} = require('./routers');
 const {walletPasswordPrompt} = require('./service');
 
 const app = express();
-const httpsPort = 18554;
-const {LNSERVICE_LND_DIR} = process.env;
 const logFormat = ':method :url :status - :response-time ms - :user-agent';
 const {NODE_ENV} = process.env;
-const port = process.env.PORT || 10553;
 const unlockDelayMs = 1000;
 const unlockerLnd = localLnd({is_unlocker: true});
 
 if (NODE_ENV !== 'production') {
   walnut.check(require('./package'));
 }
 
+module.exports = app;
```

Comment:
export the `app`.  I'm testing now the effect of doing it this way on setting up the routes in the http tests, if you have any feedback on doing this different just lmk

Reply:
does this mean it would be possible to have both ways of running the server? the bin/www way and the original way as well?

Reply:
The logic that was there before is just moved in to the `www` file to be a node script that is run when actually serving the code.

By separating it out the app can be set up independently without actually turning on the web server and listening to the port, this allows libs like `supertest` to run it and test out the end points in an integration test without having it running and calling the api manually.

HTTP tests coming in a separate PR, wanted to put this up for discussion first.

Reply:
Better explanation - 

https://stackoverflow.com/a/23249547/1200803

Reply:
Maybe it would be nicer if running nodemon server.js still does the same thing it did before but its internals change to use an abstracted server that also is used in the www/bin and for testing?

Reply:
What about -
```
$ mv ./server.js ./app.js
$ mv ./bin/www /.server.js
```
Then the command wouldn't even change?

Reply:
Updated the PR to show what I mean

Reply:
I don't 100% understand the change but if you add a test I can mess around with things and try and make sure the test still passes